h2o-r/h2o-package/R/communication.R

```diff
@@ -507,58 +507,75 @@
+        # curl 7.68.0 introduces socketpair, RCurl does not always release the sockets which leads to errors
+        if (curlVersion[[1]] >= 7 && curlVersion[[2]] >= 68) {
+          errorMessage <- paste(
+            errorMessage,
+            "\nThis can be caused by issues with some versions of curl library together with RCurl package.",
+            "Installing R package `curl` version 4.3.0 and above could help.",
+            "Otherwise, using curl system library versions below 7.68.0 or compiled with --disable-socketpair could help,",
+            "in case you cannot use curl R package."
+          )
+        }
       }
-    }
-    stop(sprintf("Unexpected CURL error: %s", errorMessage))
-  } else if (rv$httpStatusCode != 200) {
-    cat("\n")
-    cat(sprintf("ERROR: Unexpected HTTP Status code: %d %s (url = %s)\n", rv$httpStatusCode, rv$httpStatusMessage, rv$url))
-    cat("\n")
-
-    #Check if payload is a raw vector(binary data) and convert to character for error printing. Otherwise return
-    #normal payload
-    if(is.raw(rv$payload)){
-      jsonObject = jsonlite::fromJSON(rawToChar(rv$payload), simplifyDataFrame=FALSE)
-    }else{
-      jsonObject = jsonlite::fromJSON(rv$payload, simplifyDataFrame=FALSE)
-    }
-
-    exceptionType = jsonObject$exception_type
-    if (! is.null(exceptionType)) {
-      cat(sprintf("%s\n", exceptionType))
-    }
-
-    stacktrace = jsonObject$stacktrace
-    if (! is.null(stacktrace)) {
-      print(jsonObject$stacktrace)
+      stopError <- sprintf("Unexpected CURL error: %s", errorMessage)
+    } else if (rv$httpStatusCode != 200) {
+      cat("\n")
+      cat(sprintf("ERROR: Unexpected HTTP Status code: %d %s (url = %s)\n", rv$httpStatusCode, rv$httpStatusMessage, rv$url))
       cat("\n")
+    
+      #Check if payload is a raw vector(binary data) and convert to character for error printing. Otherwise return
+      #normal payload
+      if (is.raw(rv$payload)) {
+        jsonObject = jsonlite::fromJSON(rawToChar(rv$payload), simplifyDataFrame=FALSE)
+      } else {
+        jsonObject = jsonlite::fromJSON(rv$payload, simplifyDataFrame=FALSE)
+      }
+    
+      exceptionType = jsonObject$exception_type
+      if (! is.null(exceptionType)) {
+        cat(sprintf("%s\n", exceptionType))
+      }
+    
+      stacktrace = jsonObject$stacktrace
+      if (! is.null(stacktrace)) {
+        print(jsonObject$stacktrace)
+        cat("\n")
+      }
+    
+      msg = jsonObject$msg
+      if (!is.null(msg)) {
+        stopError <- msg
+      } else {
+        stopError <- "Unexpected HTTP Status code"
+      }
     }
 
-    msg = jsonObject$msg
-    if (! is.null(msg)) {
-      stop(msg)
+    if (is.null(stopError)) {
+      return(rv$payload)
+    } else if (number_of_tries <= 0) {
+      stop(stopError)
     } else {
-      stop("Unexpected HTTP Status code")
-    }
+      cat("\n")
+      cat(sprintf("Error occurred, retrying in %d seconds", retry_interval))
+      cat("\n")
```

Comment:
I'd probably use `warning` (or `message`) so it is possible to suppress it / redirect to a file etc.

Reply:
will change, thanks